TodoApp.ConsoleApp/UI/UserInput.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+
+namespace TodoApp.ConsoleApp.UI
+{
+    public static class UserInput
+    {
+        const string Yes = "yes";
+        const string No = "no";
+
+        private static string ReadInput(bool required = false, string defaultValue = "")
+        {
+            string text = "";
+            if (required)
+            {
+                text = Console.ReadLine();
+
+                while (string.IsNullOrEmpty(text))
+                {
+                    Console.WriteLine("This field can't be empty!");
+                    text = Console.ReadLine();
+                }
+            }
+
+            else
+            {
+                text = Console.ReadLine();
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    text = defaultValue;
+                }
+            }
+
+            return text;
+        }
+
+        public static string ReadText(string question, bool required = false, string defaultValue = "")
+        {
+            Console.WriteLine(question);
+            string text = "";
+            if (required)
+            {
+                text = ReadInput(required);
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    text = defaultValue;
+                }
+            }
+
+            else
+            {
+                text = defaultValue;
+            }
+
+            return text;
+        }
+
+        public static int ReadOption(string question, string[] availableOptions, bool required = false, int defaultValue = 0)
+        {
+            Console.WriteLine(question);
+            int index = 1;
+            foreach (var option in availableOptions)
+            {
+                Console.WriteLine($"\t{index}: {option}");
+                index++;
+            }
+
+            index = 1;
+            bool isValid = false;
+            string input = "";
+
+            if (required)
```

Comment:
I intentionally didn't used the `required` and `defaultValue`  in the pseudo code. `ReadInput()` uses them to validate the input.
So you don't need to check the `if-else`. Just leave the `do-while` and add both parameters when calling `ReadInput()`